### PR TITLE
[ADAM-1954] Complete refactoring GenomicRDD to GenomicDataset.

### LINKED ISSUE #1954: Clarify GenomicRDD vs. GenomicDataset name
After https://github.com/bigdatagenomics/adam/pull/1921, GenomicRDD is replaced with GenomicDataset. We should rationalize the names.

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1954.  Ready for review.

A few additional questions:

Should the `rdd` package be renamed to `dataset`?

Is it possible to move `ADAMContext` up a package, to `org.bdgenomics.adam`?  I've always wanted to do this, now might be a good time.  Not sure if it would be possible due to visibility and I wouldn't want to create any circular dependencies between packages.

### CODE REVIEW

File: adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala [hunk 1]
```diff
@@ -697,11 +697,11 @@ class ADAMContextSuite extends ADAMFunSuite {
     val sd = sc.loadSequenceDictionary(sdPath)
 
     val path = testFile("gencode.v7.annotation.trunc10.bed") // uses "chr1"
-    val featureRdd = sc.sc.loadFeatures(path, optSequenceDictionary = Some(sd))
-    val features: RDD[Feature] = featureRdd.rdd
+    val FeatureDataset = sc.sc.loadFeatures(path, optSequenceDictionary = Some(sd))
```
Comment (heuermh) [inline adam-core/src/test/scala/org/bdgenomics/adam/rdd/ADAMContextSuite.scala]: how'd this happen?

**Comment by heuermh** [comment]
Jenkins, retest this please

**Comment by antonkulaga** [comment]
Hope it will go to 0.25 , looking forward for this commit

**Comment by heuermh** [comment]
> Should the rdd package be renamed to dataset?

Per https://gitter.im/bigdatagenomics/adam?at=5b218ed5202c8f71f5ffb451, `adam.rdd` &rarr; `adam.ds` might be preferrable

**Comment by heuermh** [comment]
Rebased, ping @akmorrow13 @fnothaft @jpdna for review.

File: docs/api/genomicDataset.rst [hunk 2]
```diff
@@ -3,8 +3,8 @@ Working with genomic data using GenomicRDDs
```
Comment (jpdna) [inline docs/api/genomicDataset.rst]: Should the above be `GenomicRDDs` or `GenomicsDatasets`

Comment (jpdna) [reply docs/api/genomicDataset.rst]: Refer to line 1 of the file in 'Working with genomic data using GenomicRDDs'

File: adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordDataset.scala [hunk 3]
```diff
@@ -190,20 +190,20 @@ object AlignmentRecordRDD extends Serializable {
   def apply(ds: Dataset[AlignmentRecordProduct],
```
Comment (jpdna) [inline adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordDataset.scala]: This function could use scaladoc, the equivalent  ` def apply(ds: Dataset[GenotypeProduct],` function in `GenotypeDataset.scala` does have scaladoc and notes `..., populated from a SQL Dataset.`

**Comment by jpdna** [comment]
Other than my two small documentation nits above, this looks good to me.
I'd imagine you expect no performance changes for better or worse based on this PR, correct?

**Comment by heuermh** [comment]
@jpdna Thanks for the review.  I pushed another commit addressing your comments.

**Pull request merged**